src/validation.cpp

```diff
@@ -177,6 +178,13 @@ class CChainState {
 
     void UnloadBlockIndex();
 
+    ~CChainState() {
+        BlockMap::iterator it1 = mapBlockIndex.begin();
```

Comment:
nit: This can be simplified using the foreach loop in `UnloadBlockIndex`. And the explicit clear at the end is unnecessary.

```c++
for (BlockMap::value_type& entry : mapBlockIndex) delete entry.second;
```

Reply:
Commit "mapBlockIndex only in validation.cpp, and const outside CChainState"

> And the explicit clear at the end is unnecessary.

@jimpo why? Otherwise it would contain dangling pointers. `mapBlockIndex` isn't a `CChainState` member.